app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -201,21 +212,45 @@ class UndertakingController @Inject()(
             }
             undertakingName = UndertakingName(updatedJourney.name.value.getOrElse(handleMissingSessionData("Undertaking Name")))
             undertakingSector = updatedJourney.sector.value.getOrElse(handleMissingSessionData("Undertaking Sector"))
-            ref <- escService.createUndertaking(
-                    Undertaking(
-                      None,
-                      name = undertakingName,
-                      industrySector = undertakingSector,
-                      None,
-                      None,
-                      List(BusinessEntity(eori, leadEORI = true, updatedJourney.contact.value)
-                    )))
-
-          } yield Redirect(routes.UndertakingController.getConfirmation(ref, updatedJourney.name.value.getOrElse("")))
+            undertaking =  Undertaking(
+              None,
+              name = undertakingName,
+              industrySector = undertakingSector,
+              None,
+              None,
+              List(BusinessEntity(eori, leadEORI = true, updatedJourney.contact.value)
+              ))
+            result <- createUndertakingAndSendEmail(undertaking, eori, updatedJourney)
+
+          } yield result
 
     )
   }
 
+  //This method creates undertaking, checks for the language, fetches the appropriate template as per the lang
+  //call the retrieve email service and sends the email to retrieved email address
+  private def createUndertakingAndSendEmail(undertaking: Undertaking, eori: EORI, undertakingJourney: UndertakingJourney)(implicit request: EscAuthRequest[_]) =     for {
+    ref <- escService.createUndertaking(undertaking)
+    lang <- getLanguage
+    templateId = getTemplateId(lang)
```

Comment:
NABD - this could be one function `getTemplateId` which does the language check - I can't see it used anywhere else.

Reply:
As discussed will do as a part of net PR